core/src/main/java/io/kestra/core/test/flow/Assertion.java

```diff
@@ -22,14 +26,234 @@ public class Assertion {
     private Property<String> endsWith;
     private Property<String> startsWith;
     private Property<String> contains;
-    private Property<String> equalsTo;
-    private Property<String> notEqualsTo;
+    private Property<Object> equalsTo;
+    private Property<Object> notEqualsTo;
     private Property<Double> greaterThan;
     private Property<Double> greaterThanOrEqualTo;
     private Property<Double> lessThan;
-    private Property<Double> lesThanOrEqualTo;
+    private Property<Double> lessThanOrEqualTo;
     private Property<List<String>> in;
     private Property<List<String>> notIn;
     private Property<Boolean> isNull;
     private Property<Boolean> isNotNull;
+
+    public List<AssertionResult> run(RunContext runContext) {
```

Comment:
```suggestion
    public List<AssertionResult> evaluate(RunContext runContext) {
```

You evaluate assertions not run?
Anyway, you can keep run if you want, not important.